pkg/querier/frontend/worker.go

```diff
@@ -118,116 +111,64 @@ func (w *worker) watchDNSLoop(servCtx context.Context) error {
 			switch update.Op {
 			case naming.Add:
 				level.Debug(w.log).Log("msg", "adding connection", "addr", update.Addr)
-				ctx, cancel := context.WithCancel(servCtx)
-				cancels[update.Addr] = cancel
-				w.runMany(ctx, update.Addr)
+				client, err := w.connect(update.Addr)
+				if err != nil {
+					level.Error(w.log).Log("msg", "error connecting", "addr", update.Addr, "err", err)
+				}
+
+				w.managers[update.Addr] = newFrontendManager(servCtx, w.log, w.server, client, 0, w.cfg.GRPCClientConfig.MaxRecvMsgSize)
```

Comment:
Too many arguments to a function is a bit a smell to me.

Can you pass `w.cfg.GRPCClientConfig` and have the client create by `newFrontendManager`?

Also, probably don't need to pass 0, it can be assumed.

Reply:
Will definitely remove the 0.  I had intended that for something else, but obviously it fell by the wayside.

I would prefer continuing to pass in the client
- it would not reduce the number of required parameters.  I would have to pass in client address and client config instead of client and client config.
- it makes testing easier.  Currently I'm passing in a mock client.  If I pass in a config then I need to find a way to have the frontend manager create a real client that works for testing.

Reply:
Yeah thats a good reason to, I agree.